src/fdcache_page.cpp

```diff
@@ -835,17 +836,26 @@ bool PageList::Serialize(CacheFileStat& file, ino_t inode)
     for(auto iter = pages.cbegin(); iter != pages.cend(); ++iter){
         ssall << "\n" << iter->offset << ":" << iter->bytes << ":" << (iter->loaded ? "1" : "0") << ":" << (iter->modified ? "1" : "0");
     }
+    std::string strall = ssall.str();
 
+    if(0 != flock(file.GetFd(), LOCK_EX)){
```

Comment:
Let's not do file locking.  First of all this is advisory and it would require the shell script to use `flock(1)`.  But second this is slower.  Instead could s3fs write to a temporary file and rename on top of the old one?

Reply:
Changed codes for using rename function instead of flock.